### ISSUE TITLE
TypeError: argument of type 'NoneType' is not iterable

### ISSUE THREAD

**Issue by talhaanwarch** [awslabs/autogluon#1109]
My data has two columns, `[image,labels]` and I load images as follow
df = ImagePredictor.Dataset.from_csv('train.csv', root='train_images')
when I do `df.show_images()` it plot the images. But when I do 
```
predictor = ImagePredictor()
predictor.fit(df, hyperparameters={'epochs': 2})
```
It gives following  error 
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-39-e0442c33015a> in <module>
      1 predictor = ImagePredictor()
----> 2 predictor.fit(df, hyperparameters={'epochs': 2})  # you can trust the default config, we reduce the # epoch to save some build time

/opt/conda/lib/python3.7/site-packages/autogluon/vision/configs/presets_configs.py in _call(*args, **kwargs)
     13         def _call(*args, **kwargs):
     14             gargs, gkwargs = set_presets(preset_name, *args, **kwargs)
---> 15             return f(*gargs, **gkwargs)
     16         return _call
     17     return _unpack_inner

/opt/conda/lib/python3.7/site-packages/autogluon/vision/predictor/predictor.py in fit(self, train_data, tuning_data, time_limit, presets, hyperparameters, **kwargs)
    293 
    294         # data sanity check
--> 295         train_data = self._validate_data(train_data)
    296         train_labels = _get_valid_labels(train_data)
    297         self._label_cleaner = LabelCleaner.construct(problem_type=self._problem_type, y=train_labels, y_uncleaned=train_labels)

/opt/conda/lib/python3.7/site-packages/autogluon/vision/predictor/predictor.py in _validate_data(self, data)
    414                 raise TypeError(f"Unable to process dataset of type: {type(data)}")
    415         elif isinstance(data, _ImageClassification.Dataset):
--> 416             assert 'label' in data.columns
    417             assert hasattr(data, 'classes')
    418             # check whether classes are outdated, no action required if all unique labels is subset of `classes`

AssertionError: 
```
which mean that column name should be `label` instead of `labels`
so when I changed column name with `label` as follow
```
df = ImagePredictor.Dataset.from_csv('train.csv', root='train_images')
df.columns=['image','label']
```
and do `df.show_images()` I got following error
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-41-741ee22ad2e8> in <module>
----> 1 df.show_images()

/opt/conda/lib/python3.7/site-packages/gluoncv/auto/data/dataset.py in show_images(self, indices, nsample, ncol, shuffle, resize, fontsize)
    132         if 'label' in self.columns:
    133             titles = [self.classes[int(self.at[idx, 'label'])] + ': ' + str(self.at[idx, 'label']) \
--> 134                 for idx in indices if idx < len(self)]
    135         _show_images(images, cols=ncol, titles=titles, fontsize=fontsize)
    136 

/opt/conda/lib/python3.7/site-packages/gluoncv/auto/data/dataset.py in <listcomp>(.0)
    132         if 'label' in self.columns:
    133             titles = [self.classes[int(self.at[idx, 'label'])] + ': ' + str(self.at[idx, 'label']) \
--> 134                 for idx in indices if idx < len(self)]
    135         _show_images(images, cols=ncol, titles=titles, fontsize=fontsize)
    136 

ValueError: invalid literal for int() with base 10: 'frog_eye_leaf_spot'
```
and 
```
predictor = ImagePredictor()
predictor.fit(df, hyperparameters={'epochs': 2})
```
gives following error

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-42-e0442c33015a> in <module>
      1 predictor = ImagePredictor()
----> 2 predictor.fit(df, hyperparameters={'epochs': 2})  # you can trust the default config, we reduce the # epoch to save some build time

/opt/conda/lib/python3.7/site-packages/autogluon/vision/configs/presets_configs.py in _call(*args, **kwargs)
     13         def _call(*args, **kwargs):
     14             gargs, gkwargs = set_presets(preset_name, *args, **kwargs)
---> 15             return f(*gargs, **gkwargs)
     16         return _call
     17     return _unpack_inner

/opt/conda/lib/python3.7/site-packages/autogluon/vision/predictor/predictor.py in fit(self, train_data, tuning_data, time_limit, presets, hyperparameters, **kwargs)
    293 
    294         # data sanity check
--> 295         train_data = self._validate_data(train_data)
    296         train_labels = _get_valid_labels(train_data)
    297         self._label_cleaner = LabelCleaner.construct(problem_type=self._problem_type, y=train_labels, y_uncleaned=train_labels)

/opt/conda/lib/python3.7/site-packages/autogluon/vision/predictor/predictor.py in _validate_data(self, data)
    418             # check whether classes are outdated, no action required if all unique labels is subset of `classes`
    419             unique_labels = sorted(data['label'].unique().tolist())
--> 420             if not (all(ulabel in data.classes for ulabel in unique_labels)):
    421                 data = _ImageClassification.Dataset(data, classes=unique_labels)
    422                 logger.log(20, f'Reset labels to {unique_labels}')

/opt/conda/lib/python3.7/site-packages/autogluon/vision/predictor/predictor.py in <genexpr>(.0)
    418             # check whether classes are outdated, no action required if all unique labels is subset of `classes`
    419             unique_labels = sorted(data['label'].unique().tolist())
--> 420             if not (all(ulabel in data.classes for ulabel in unique_labels)):
    421                 data = _ImageClassification.Dataset(data, classes=unique_labels)
    422                 logger.log(20, f'Reset labels to {unique_labels}')

TypeError: argument of type 'NoneType' is not iterable
```
I tried using label encoder but the problem is still there though error changes

**Comment by talhaanwarch** [comment]
solution is that we need to change label of csv file first, then save it and then load it again. also need to perform encoding

**Comment by talhaanwarch** [comment]
i also tried to specify column name, but it did not work 
predictor = ImagePredictor(label="labels")

**Comment by talhaanwarch** [comment]
Here is[ colab notebook ](https://colab.research.google.com/drive/1UJw6Hvl7B-B9gdH5yMKt0jmY8bJt_wHa?usp=sharing)

**Comment by zhreshold** [comment]
@talhaanwarch see fix in #1113, the root cause is that the label column is not properly recognized in Dataset, causing the sequential problems. 
You can update autogluon once the PR is merged, it will automatically fix the issue, or you can use the raw DataFrame instead.

For example, instead of 
```python
df = ImagePredictor.Dataset.from_csv('/content/train.csv', root='/content/train_images')
# this should be fixed in the future to allow custom label column as well
```
use 
```python
import os
import pandas as pd
df  = pd.read_csv('/content/train.csv')
df['image'] = df['image'].apply(lambda x: os.path.join('/content/train_images', x))
df.head()
```

BTW, you are installing `mxnet<2.0.0`, which is cpu only wheel. You can install `mxnet-cu110<2.0.0` instead to enable gpu on colab.
